### PR TITLE
[fix-6766][api]fix bug when query projectList by api

### LINKED ISSUE #6766: [Bug] [API] queryAllProjectList response userName is null
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

api ：queryAllProjectList response userName is null
image

The API interface with the same condition has 
“queryAuthorizedProject”
“queryProjectCreatedAndAuthorizedByUser”
“queryUnauthorizedProject” 

### What you expected to happen

The response return from api with correct userName
image

### How to reproduce

use the api to query project ：“queryAllProjectList",“queryAuthorizedProject”,“queryProjectCreatedAndAuthorizedByUser”,“queryUnauthorizedProject” 

### Anything else

Version：2.0.0-alpha-release

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xianweiY**
/dolphinscheduler/projects/authed-project : fix missing project infos such as  'userName','defCount','instRunningCount'
/dolphinscheduler/projects/created-and-authed: fix missing project infos such as  'userName','defCount','instRunningCount'
/dolphinscheduler/projects/list : Fix permission bug, any user can query all projects before, now user can only query authorized  project or create himself
/dolphinscheduler/projects/list: fix missing project infos such as   'userName','defCount','instRunningCount'
/dolphinscheduler/projects/{code} : fix missing project infos such as   'userName','defCount','instRunningCount'

complete unitTest

this close #6766

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by xianweiY** [comment]
image

**Review by CalvinKirs — COMMENTED** [review]
The back-end code looks good. SQL needs to come alone to REVIEW

**Comment by xianweiY** [comment]
> The back-end code looks good. SQL needs to come alone to REVIEW

What should I do, or wait for the  SQL review to complete?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectController.java [hunk 1]
```diff
@@ -265,7 +265,7 @@ public Result queryProjectCreatedAndAuthorizedByUser(@ApiIgnore @RequestAttribut
     @ApiException(LOGIN_USER_QUERY_PROJECT_LIST_PAGING_ERROR)
     @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
     public Result queryAllProjectList(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser) {
-        Map<String, Object> result = projectService.queryAllProjectList();
+        Map<String, Object> result = projectService.queryAllProjectList(loginUser);
```
Comment (lenboo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectController.java]: there are some situations that we need the all projects without login user.
so i think this is no necessary.

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.java [hunk 2]
```diff
@@ -114,7 +114,7 @@
     ProjectUser queryProjectWithUserByProcessInstanceId(@Param("processInstanceId") int processInstanceId);
 
     /**
-     * query all project
+     * query all project,only admin user can use this method, Other users query all projects use {@link ProjectMapper#queryProjectCreatedAndAuthorizedByUserId(int)}
```
Comment (lenboo) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.java]: not only admin can use this api.

**Review by lenboo — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectController.java (hunk 1 above)
Comment (xianweiY) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectController.java]: If this interface has no permission control, then you are right and this is no necessary, I have removed it in the new commit

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.java (hunk 2 above)
Comment (xianweiY) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.java]: if now anyone can use this method to query all projects , this comment is wrong, i have remove

**Pull request closed without merging**